concepts/class-inheritance/about.md

```diff
@@ -1,2 +1,291 @@
+     __div = None
+     
+     # constructor
+    def __init__(self, name, roll, marks, div): 
+        self.__name = name
+        self.__roll = roll
+        self.__marks = marks
+        self.__div = div
+     
+     # public member function 
+     def displayDetails(self):
+         # accessing private data members
+         print("Name: ", self.__name)
+         print("Roll: ", self.__roll)
+         print("Marks:", self.__marks)
+         print("Division: ", self.__div)
+         
+     # Accessor Function    
+     def getRollNo(self):
+        return self.__roll
+     
+     # Mutator Function
+     def setDiv(self):
+        if self.__marks>=60:
+            self.__div = 'A'
+        elif self.__marks<60:
+            self.__div = 'B'
+        return self.__div
+```
+__2. Mutator Function__: This method is used to change the state of an object, i.e. it changes the data variable's hidden value. It has the ability to change the value of a variable on the fly. The update method is another name for this function. We can also use the word _set_ to name these procedures.
+
+As per the above example, `getRollNo()` is an accessor function and `setDiv` is a mutator function.
+
+## Inheritance
+
+## Single Inheritance
+
+When a subclass inherits only from one base class, it is known as single inheritance.
+
+```python
+#Base class
+class Odin:
+    def func1(self):
+        print("Odin: father of thor")
+
+#Derived/Subclass class
+class Thor(Odin):
+    def func2(self):
+        print("Thor: son of Odin")
+
+
+object = Thor()
+object.func1()
+object.func2()
+```
+The derived class will print both the statements of `func1` and `func2`.
+
+## Multiple Inheritance
+As we've seen, a class inherits from another class in this scenario. On the other side, multiple inheritance is a feature that allows a class to inherit characteristics and methods from many parent classes.
+```python
+class SubclassName(BaseClass1, BaseClass2, ...):
+    pass
+```
+### Diamond Problem
+The "diamond problem" (also known as the "deadly diamond of death") refers to an ambiguity that occurs when two classes B and C inherit from a superclass A, while another class D inherits from both B and C. If A has a method "m" that B or C (or even both of them) has overridden, and if it does not override this method, the question becomes which version of the method D inherits. It's possible that it's from A, B, or C.
+Let's have a look at the problem using an example:
+
+```python
+class A:
+    def m(self):
+        print("m of A called")
+class B(A):
+    def m(self):
+        print("m of B called")
+class C(A):
+    def m(self):
+        print("m of C called")
+class D(B,C):
+    pass
+```
+If we call an instance x of class D, we will get the output as `m of B called`. But if we interchange the order of inheritance in class D i.e. `Class D(C, D)`. We will get the output as `m of C called`. 
+To solve the diamond problem in python, we will look into a new method `mro()`.

```

Comment:
This isn't a new method, this is how Python solves the diamond problem.  `mro()` exposes what Python is doing, but it doesn't turn it on or off.  Suggest wording closer to what is [here](https://www.educative.io/edpresso/what-is-mro-in-python)